File: Integraupt-Backend/BackendAdReserva/src/main/java/com/AdReserva/repository/EspacioRepository.java
package com.AdReserva.repository;

import com.AdReserva.model.Espacio;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;

import java.util.List;

public interface EspacioRepository extends JpaRepository<Espacio, Integer> {

    @Query(value = "SELECT DISTINCT Tipo FROM espacio ORDER BY Tipo", nativeQuery = true)
    List<String> obtenerTiposDeEspacio();
}

File: Integraupt-Backend/servicio-adreserva/src/Repositories/EspacioRepository.php
<?php

namespace App\Repositories;

use App\Core\Database;
use PDO;

class EspacioRepository {
    private PDO $db;

    public function __construct() {
        $this->db = Database::getConnection();
    }

    /**
     * @return string[]
     */
    public function obtenerTiposDeEspacio(): array {
        $sql = "SELECT DISTINCT Tipo FROM espacio ORDER BY Tipo";
        $stmt = $this->db->query($sql);
        
        $tipos = [];
        while ($row = $stmt->fetch()) {
            if ($row['Tipo'] !== null && trim($row['Tipo']) !== '') {
                $tipos[] = $row['Tipo'];
            }
        }
        
        return $tipos;
    }
}
